jdaviz/core/template_mixin.py

```diff
@@ -4401,22 +4401,16 @@ def is_mixed(self, glue_values):
         return len(np.unique(glue_values, axis=0)) > 1
 
     def _update_mixed_state(self):
-        if len(self.linked_states) <= 1:
-            mixed = False
-        else:
+
+        if len(self.linked_states) >= 1:
             current_glue_values = []
             for state in self.linked_states:
                 current_glue_values.append(self._get_glue_value(state))
-            mixed = self.is_mixed(current_glue_values)
-            # ensure the value corresponds to the first entry, this prevents the case where a glue
-            # change to one of the linked_states changes the value that will be adopted when
-            # unmixing something in mixed state and results in more consistent and predictable
-            # behavior
-            self._processing_change_from_glue = True
-            self.value = current_glue_values[0]
-            self._processing_change_from_glue = False
-        self.sync = {**self.sync,
-                     'mixed': mixed}
+
+        if len(current_glue_values) and current_glue_values[0] is not None:
+            # sync the initial value of the widget, avoiding recursion
+            self._on_glue_value_changed(current_glue_values[0],
+                                        update_mixed_state=False)
```

Comment:
this still needs all the old logic.  I think all the changes here can be reverted now and just replace:

```
self.value = current_glue_values[0]
```

with

```
self._on_glue_value_changed(current_glue_values[0], update_mixed_state=False)
```

the changes you made to `_on_glue_value_changed` will then avoid the recursion.